source/mesh_deformation/interface.cc

```diff
@@ -722,6 +740,253 @@ namespace aspect
 
 
 
+    template <int dim>
+    void MeshDeformationHandler<dim>::compute_mesh_displacements_gmg()
+    {
+      // Same as compute_mesh_displacements, but using GMG instead of AMG.
+
+      // We use this gmg solver only when the gmg stokes solver is used
+      // for the following reasons (TODO):
+      // 1. this gmg solver does not support periodic boundary conditions
+      // 2. To use this solver even when gmg stokes solver is not used, we need to
+      //    initialize the triangulation with Triangulation<dim>::limit_level_difference_at_vertices
+      //    and parallel::distributed::Triangulation<dim>::construct_multigrid_hierarchy
+      // 3. Although this gmg solver is much faster than the amg solver, it's only tested for
+      //    limited free surface cases.
+
+      TimerOutput::Scope timer (this->sim.computing_timer, "gmg displacement");
```

Comment:
The other `compute_mesh_displacements` function does not have such a timer section, and both function are embedded in another timer section (`Mesh deformation initialize`), which makes the output confusing (because the total does add up to more than 100% because of the duplication). I understand this was useful for benchmarking, but I would prefer you remove this timer section for the merge.